spec/suites/geo/LatLngSpec.js

```diff
@@ -2,21 +2,51 @@ describe('LatLng', function () {
 	describe('constructor', function () {
 		it("sets lat and lng", function () {
 			var a = new L.LatLng(25, 74);
-			expect(a.lat).to.eql(25);
-			expect(a.lng).to.eql(74);
+			expect(a.lat).to.be(25);
+			expect(a.lng).to.be(74);
 
 			var b = new L.LatLng(-25, -74);
-			expect(b.lat).to.eql(-25);
-			expect(b.lng).to.eql(-74);
-		});
-
-		it('throws an error if invalid lat or lng', function () {
-			expect(function () {
-				L.latLng(NaN, NaN);
-			}).to.throwError();
-		});
-
-		it('does not set altitude if undefined', function () {
+			expect(b.lat).to.be(-25);
+			expect(b.lng).to.be(-74);
+		});
+
+		it("convert from string", function () {
+			var a = new L.LatLng("25", " 74");
+			expect(a.lat).to.be(25);
+			expect(a.lng).to.be(74);
+
+			var b = new L.LatLng("-25", "-74 ");
+			expect(b.lat).to.be(-25);
+			expect(b.lng).to.be(-74);
+
+			var c = new L.LatLng("010", "0x10"); // better'd throw here..
+			expect(c.lat).to.be(10);
+			expect(c.lng).to.be(0); // bad!
+		});
+
+		it('throws an error if invalid lat, lng, alt', function () {
+			expect(L.latLng).withArgs(0, 0, undefined).to.throwError();
+			expect(L.latLng).withArgs(0).to.throwError();
+			expect(L.latLng).withArgs(NaN, 0).to.throwError();
+			expect(L.latLng).withArgs(Infinity, 0).to.throwError();
+			expect(L.latLng).withArgs(-Infinity, 0).to.throwError();
+			expect(L.latLng).withArgs(true, 0).to.throwError();
+			expect(L.latLng).withArgs(false, 0).to.throwError();
+			expect(L.latLng).withArgs(undefined, 0).to.throwError();
+			expect(L.latLng).withArgs(null, 0).to.throwError();
+			expect(L.latLng).withArgs('', 0).to.throwError();
+			expect(L.latLng).withArgs('\n', 0).to.throwError();
+			expect(L.latLng).withArgs('1px', 0).to.throwError();
+			expect(L.latLng).withArgs('nonsense', 0).to.throwError();
+			expect(L.latLng).withArgs([]).to.throwError();
+			expect(L.latLng).withArgs([50]).to.throwError();
+			// do not throw atm:
+			// expect(L.latLng).withArgs('0010', 0).to.throwError();
+			// expect(L.latLng).withArgs('0x10', 0).to.throwError();
+			// expect(L.latLng).withArgs('1e5', 0).to.throwError();

```

Comment:
And here.

Reply:
`1e5` = `100000` is a correct number, what is not vaild in my eyes is that the string `'1e5'` is converted to `1` (`parseInt('1e5')`)

Reply:
> is a correct number

Sure, but do we really expect such number in Leaflet?

Reply:
Maybe in CRSSimple but I don't think so. I think we should only check if it is a valid number and not if the number makes sense